protocol/xmrtaker/swap_state.go

```diff
@@ -484,17 +487,34 @@ func (s *swapState) tryRefund() (ethcommon.Hash, error) {
 	// it won't handle those events while this function is executing.)
 	log.Infof("waiting until time %s to refund", s.t1)
 
-	event := <-s.eventCh
-	log.Debugf("got event %s while waiting for T1", event)
-	switch event.(type) {
-	case *EventShouldRefund:
+	waitCtx, waitCtxCancel := context.WithCancel(context.Background())
+	defer waitCtxCancel() // Unblock WaitForTimestamp if still running when we exit
+
+	waitCh := make(chan error)
+	go func() {
+		waitCh <- s.ETHClient().WaitForTimestamp(waitCtx, s.t1)
+		close(waitCh)
+	}()
+
+	select {
+	case event := <-s.eventCh:
+		log.Debugf("got event %s while waiting for T1", event)
+		switch event.(type) {
+		case *EventShouldRefund:
+			return s.refund()
+		case *EventETHClaimed:
+			// we should claim; returning this error
+			// causes the calling function to claim
+			return ethcommon.Hash{}, fmt.Errorf(revertSwapCompleted)
+		default:
+			panic(fmt.Sprintf("got unexpected event while waiting for Claimed/T1: %s", event))
+		}
+	case err = <-waitCh:
+		if err != nil {
+			panic(fmt.Sprintf("failed to wait for T1: %s", err))
```

Comment:
Same thing here. This seems like something that can legitimately happen without any bugs in our code.

Reply:
yeah, I removed the panics. I guess my thought process was that if this case is hit, it's really bad because the swap is unable to complete. if this happens it's probably due to bad endpoint or some other reason though, and the user would need to restart anyways